webtraversallibrary/workflow.py

```diff
@@ -230,59 +234,68 @@ def _get_new_views(self) -> Dict[str, View]:
                 all_views[tab] = self.history[-1]
                 continue
 
-            # Ensure page is fully loaded
-            self.current_window.scraper.wait_until_loaded()
+            # Store View
+            all_views[tab] = self._get_new_view(tab, initial_action)
+
+        return all_views
+
+    def _get_new_view(self, name: str, initial_action: Action) -> View:
+        # Ensure page is fully loaded
+        self.current_window.scraper.wait_until_loaded()
 
-            # Run postload callbacks
-            for cb in self.current_window.scraper.postload_callbacks:
-                cb()
+        # Run postload callbacks
+        for cb in self.current_window.scraper.postload_callbacks:
+            cb()
 
-            # Scrape the page
-            snapshot = self.current_window.scraper.scrape_current_page()
+        # Scrape the page
+        snapshot = self.current_window.scraper.scrape_current_page()
 
-            # Assemble basic list of actions
-            action_list: List[Action] = [Abort(), Refresh(), Navigate(), Wait()]
-            action_list += [Revert(step) for step in range(len(self.history))]
-            actions = Actions(action_list)
+        # Assemble basic list of actions
+        action_list: List[Action] = [Abort(), Refresh(), Navigate(), Wait()]
+        action_list += [Revert(step) for step in range(len(self.history))]
+        actions = Actions(action_list)
 
-            # Setup metadata
+        # Setup metadata
+        metadata: Dict[str, Any] = {}
+        if self.config.scraping.history:
             if self.history and "next_action" in self.history[self.loop_idx - 1].metadata:
                 metadata = self.history[self.loop_idx - 1].metadata.copy()
                 metadata["previous_action"] = metadata["next_action"]
             else:
-                metadata = {}
                 metadata["previous_action"] = [initial_action]
-            metadata["next_action"] = None
+        metadata["next_action"] = None
 
-            # Create view
-            view = View(name=tab, snapshot=snapshot, actions=actions, metadata=metadata)
+        # Create view
+        view = View(name=name, snapshot=snapshot, actions=actions, metadata=metadata)
 
-            # Add state to history
-            if self.loop_idx < len(self.history):
-                self.history[self.loop_idx] = view
-            else:
-                self.history.append(view)
+        # Add state to history
+        if self.loop_idx < len(self.history):
+            self.history[self.loop_idx] = view
+        else:
+            self.history.append(view)
 
-            # Run element classifiers
-            view.actions.extend(self._run_element_classifiers(snapshot))
+        # Maintain only one level of history if required
+        if not self.config.scraping.history:
+            for i in range(len(self._history[self.current_tab]) - 1):
+                self._history[self.current_tab][i] = None
 
```

Comment:
Is there a reason why we keep the list of the same length but stack it with `None` instead of just removing the elements?

Reply:
Yeah, otherwise we'll need some special cases where other code looks for the view at the current loop_index instead of the last one. Shouldn't have any negative impact doing it this way.